src/longbow/sift/command.py

```diff
@@ -0,0 +1,208 @@
+@click_log.simple_verbosity_option(logger)
+@click.option(
+    "-p",
+    "--pbi",
+    required=False,
+    type=click.Path(exists=True),
+    help="BAM .pbi index file",
+)
+@click.option(
+    "-o",
+    "--output-bam",
+    default="-",
+    type=click.Path(exists=False),
+    help="filtered bam output (passing reads only)  [default: stdout]",
+)
+@click.option(
+    "-x",
+    "--reject-bam",
+    default="/dev/null",
+    type=click.Path(exists=False),
+    help="Filtered bam output (failing reads only)  [default: /dev/null]",
+)
+@click.option(
+    "-m",
+    "--model",
+    help="The model to use for annotation.  If not specified, it will be autodetected from "
+         "the BAM header.  If the given value is a pre-configured model name, then that "
+         "model will be used.  Otherwise, the given value will be treated as a file name "
+         "and Longbow will attempt to read in the file and create a LibraryModel from it.  "
+         "Longbow will assume the contents are the configuration of a LibraryModel as per "
+         "LibraryModel.to_json()."
+)
+@click.option(
+    "-v",
+    "--validation-model",
+    default="10x_sc_10x5p_single_none",
+    help="The model to use for cDNA validation."
+)
+@click.option(
+    '-f',
+    '--force',
+    is_flag=True,
+    default=False,
+    show_default=True,
+    help="Force overwrite of the output files if they exist."
+)
+@click.option(
+    "-s",
+    "--stats",
+    default="/dev/null",
+    type=click.Path(exists=False),
+    help="Table describing the ways in which the reads do not conform to expectation (failing reads only)  [default: /dev/null]",
+)
+@click.argument("input-bam", default="-" if not sys.stdin.isatty() else None, type=click.File("rb"))
+def main(pbi, output_bam, reject_bam, model, validation_model, force, stats, input_bam):
+    """Filter segmented reads by conformation to expected cDNA design."""
+
+    t_start = time.time()
+
+    logger.info("Invoked via: longbow %s", " ".join(sys.argv[1:]))
+
+    pbi = f"{input_bam.name}.pbi" if pbi is None else pbi
+    read_count = None
+    if os.path.exists(pbi):
+        read_count = bam_utils.load_read_count(pbi)
+        logger.info("About to Filter %d reads", read_count)
+
+    # Check to see if the output files exist:
+    bam_utils.check_for_preexisting_files([output_bam, reject_bam], exist_ok=force)
+
+    logger.info(f"Writing reads that conform to the model to: {output_bam}")
+    logger.info(f"Writing reads that do not conform to the model to: {reject_bam}")
+
+    # Open our input bam file:
+    pysam.set_verbosity(0)
+    with pysam.AlignmentFile(input_bam, "rb", check_sq=False, require_index=False) as bam_file, \
+            tqdm.tqdm(
+            desc="Progress",
+            unit=" read",
+            colour="green",
+            file=sys.stderr,
+            disable=not sys.stdin.isatty(),
+            total=read_count
+            ) as pbar:
+
+        # Get our model:
+        if model is None:
```

Comment:
For now, since this only works for one model, can you add a check to assert that the given model is the one that works with this tool?